_posts/2016-03-16-code-smells-part-I.md

```diff
@@ -0,0 +1,259 @@
+---
+layout: post
+name: code-smells-part-one
+title: Code Smells – Part I
+date: 2016-03-16 12:10:00 +00:00
+author: Ana Nogal
+canonical:
+    name: my personal blog
+    href: http://www.ananogal.com/blog/code-smells-part-one/
+image:
+    src: /assets/img/custom/blog/code-smells.jpg
+    attribution:
+        text: Via Flickr/Creative Commons/Brian Fitzgerald
+tags:
+- bloaters
+- code smells
+- long classes/methods
+---
+Last weekend I was at the [SoCraTes Canaries](https://twitter.com/hashtag/socracan16) and I gave my first talk ever about code smells...Oh boy! How nervous was I! But now that that has passed I was wondering what I should do with all information I collected: and then I thought, maybe it's a good idea to put it all in a nice blog post.
+
+## So what are code smells?
+
+As [Martin Fowler](http://martinfowler.com/) said in his book ["Refactoring: Improving the Design of Existing Code"](http://martinfowler.com/books/refactoring.html),
+> **A code smell is a surface indication that usually corresponds to a deeper problem in the system.**
+
+I like to think that a  code smell is something that makes your developer instinct cry out to you, and you just know that something is wrong. This doesn’t mean you have to make changes in your code: there are occasions where these code smells are ok, but I think It’s important for us to detect them and know exactly why they are there.
+There are 5 categories in code smells:
+
+- Bloaters
+- Object-Orientation Abusers
+- Change Preventers
+- Dispensables
+- Couplers
+
+Today I'm going to talk about Bloaters. The other categories I let then for a future post.
+
+## Bloaters
+
+Bloaters can be big methods or classes, primitive obsessions, data clumps, or long parameters list.
+
+#### Long Parameter List/Data Clumps
+
+The Long Parameter List is when you have a method that has more than 3 parameters. Sometimes we see that when we receive an object, and instead of passing it all we pass some of its data. In this case, the best policy is to pass the whole object. Data Clumps are a bit different: they are, in general, primitive values that start do "get together". A good example of this is a startDate and endDate... Maybe it's worth creating a DateRange.
```

Comment:
they are, in general, primitive values that start to "get together". A good example of this is a startDate and endDate. Maybe it's worth creating a DateRange.